### DISCUSSION TITLE
baseColorFactor when no texture?

### DISCUSSION THREAD

**Discussion by blitz-research** [KhronosGroup/glTF#2443] [Q&A]
Hi,

The spec says to treat baseColorFactor as a linear color when there is a baseColorTexture also present, ie:

"If both factors and textures are present, the factor value acts as a linear multiplier for the corresponding texture value"

but it doesn't say how to treat baseColorFactor if a baseColorTexture is not present. Should it be treated as a linear or non-linear color?

I'm guessing linear as well, but I can also see it we be convenient if it were non-linear too, so just wanted to clarify.

Bye!
Mark

**Comment by blitz-research** [comment]
Oops, I appear to have created an issue from discussion - didn't mean too, cool confetti effect though!

**Comment by lexaknyazev** [comment]
Good catch, this is actually an issue given the current spec language. It shouldn't say "If both factors and textures are present" because all factors present in glTF JSON are linear.

Filed #2444.
